frontend/libs/studio-components/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -264,6 +242,42 @@ describe('StudioCodeListEditor', () => {
     await user.type(validValueInput, newValue);
     expect(onInvalid).not.toHaveBeenCalled();
   });
+
+  it('Should save all values as numbers, when a code list that contains numbers is modified', async () => {
+    const user = userEvent.setup();
+    renderCodeListEditor({ codeList: codeListWithNumbers });
+
+    const valueInput = screen.getByRole('textbox', { name: texts.itemValue(1) });
+    const newValue = '3.1416';
+    await user.type(valueInput, newValue);
+
+    expect(onChange).toHaveBeenCalledTimes(newValue.length);
+    expect(onChange).toHaveBeenLastCalledWith([
+      { ...codeListWithNumbers[0], value: Number(newValue) },
```

Comment:
Når vi tester konvertering av typer, er det en fordel å skrive verdien rett ut. Her brukes `Number()` både i testen og i funksjonen som testes, så hvis noe er feil i forbindelse med nettopp `Number`-funksjonen, vil ikke testen oppfatte feilen.
```suggestion
      { ...codeListWithNumbers[0], value: 3.1416 },
```